client/models.py

```diff
@@ -84,6 +89,37 @@ class Volunteer(Timestampable):
         ("AGRICULTURE", "חקלאות")
     )
 
+    _original_values = {
+        'first_name': None,
+        'last_name': None,
+        'tz_number': None
+    }
+
+    def _update_original_values(self):
+        """
+        Updates the dict used to keep track of changes in certain fields between changes
+        """
+        self._original_values['tz_number'] = self.tz_number
+        self._original_values['first_name'] = self.first_name
+        self._original_values['last_name'] = self.last_name
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._update_original_values()
+
+    def save(self, *args, **kwargs):
+        if self.id is not None:
+            for key, value in self._original_values.items():
+                if getattr(self, key) != value:
+                    # A change occurred in one of the fields listed in the dict - update valid certificates images
+                    for certificate in self.get_active_certificates():
+                        client.certificate_manager.create_certificate_image(certificate)
+
+                    break
+
+        super().save(*args, **kwargs)
+        self._update_original_values()
+
```

Comment:
is this related to the issue?

Reply:
& I didn't understand what did you attempted to do here

Reply:
Yes: In order to check if one of the cruical fields (tz_number, first_name, last_name) changed, I'm storing the original value in the field, and before each save I check if the current value is the same as the original one. If it isn't, it means that the field change and the certificates should be reloaded. After this I re-update the original values so it will work again on the next save :)